src/parser/monk/mistweaver/modules/spells/EnvelopingMists.js

```diff
@@ -23,55 +22,45 @@ class EnvelopingMists extends Analyzer {
   };
 
   healingIncrease = 0;
+  evmHealingIncrease = 0;
   gustsHealing = 0;
   lastCastTarget = null;
   numberToCount = 0;
 
   constructor(...args) {
     super(...args);
     this.evmHealingIncrease = this.selectedCombatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id) ? .4 : .3;
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.ENVELOPING_MIST), this.castEnvelopingMist);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.handleEnvelopingMist);
   }
 
-  on_byPlayer_cast(event) {
-    const spellId = event.ability.guid;
-
-    if (SPELLS.ENVELOPING_MIST.id !== spellId) {//bail early if not the right spell
-      return;
-    }
+  castEnvelopingMist(event) {
     this.numberToCount += 1;
     this.lastCastTarget = event.targetID;
+  
   }
 
-  on_byPlayer_heal(event) {
+  handleEnvelopingMist(event) {
     const targetId = event.targetID;
     const spellId = event.ability.guid;
-
+    
     if (UNAFFECTED_SPELLS.includes(spellId)) {
-      debug && console.log('Exiting');
       return;
     }
 
-    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === event.targetID) && this.numberToCount >0) {
+    if ((spellId === SPELLS.GUSTS_OF_MISTS.id) && (this.lastCastTarget === targetId) && this.numberToCount >0) {
       this.gustProc += 1;
       this.gustsHealing += (event.amount || 0) + (event.absorbed || 0);
       this.numberToCount -= 1;
     }
-
+    
     if (this.combatants.players[targetId]) {
       if (this.combatants.players[targetId].hasBuff(SPELLS.ENVELOPING_MIST.id, event.timestamp, 0, 0) === true) {
         this.healingIncrease += calculateEffectiveHealing(event, this.evmHealingIncrease);
-        debug && console.log('Event Details for Healing Increase: ' + event.ability.name);
       }
     }
   }
-
-  on_fightend() {
-    if (debug) {
-      console.log(`EvM Healing Contribution: ${this.healingIncrease}`);
-      console.log(`EnM Boost`, this.evmHealingIncrease);
-      console.log("gusts env healing: ", this.gustsHealing);
-    }
-  }
+ 
 
   statistic() {
     return (
```

Comment:
I can't flag it directly and its not vital to this pr but we need to go back and update all StatisticBoxes to be Statistics instead.